uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/grids/impl/BaseGridRenderer.java

```diff
@@ -439,4 +451,124 @@ public boolean onGroupingToggle(double cellX,
+        return highlightCellRowIndex;
+    }
+
+    private List<RendererCommand> renderHighlightedCells(final GridData model,
+                                                         final GridBodyRenderContext context,
+                                                         final BaseGridRendererHelper rendererHelper,
+                                                         final BaseGridRendererHelper.RenderingInformation renderingInformation) {
+
+        final List<RendererCommand> commands = new ArrayList<>();
+
+        final GridColumn<?> column = model.getColumns().get(getHighlightCellColumnIndex());
+
+        final int visibleRowIndex = getHighlightCellRowIndex() - renderingInformation.getMinVisibleRowIndex();
+
+        commands.add((rc) -> {
+            if (!rc.isSelectionLayer()) {
+                if (model.getColumns().size() >= 1) {
+
+                    rc.getGroup().add(makeCellHighlight(getHighlightCellRowIndex(),
+                                                        visibleRowIndex,
+                                                        model,
+                                                        rendererHelper,
+                                                        column,
+                                                        context));
+                }
+            }
+        });
+
+        return commands;
+    }
+
+    Rectangle makeCellHighlight(final int rowIndex,
+                                final int visibleRowIndex,
+                                final GridData model,
+                                final BaseGridRendererHelper rendererHelper,
+                                final GridColumn<?> column,
+                                final GridBodyRenderContext context) {
+
+        final Rectangle r = getTheme().getHighlightedCellBackground().setListening(false);
+        setCellHighlightX(r, context, rendererHelper);
+        setCellHighlightY(r, rendererHelper, visibleRowIndex, model);
+        setCellHighlightSize(r, model, column, rowIndex);
+        return r;
+    }
+
+    void setCellHighlightY(final Rectangle r,
+                           final BaseGridRendererHelper rendererHelper,
+                           final int visibleRowIndex,
+                           final GridData model) {
+
+        final GridCell<?> cell = model.getCell(getHighlightCellRowIndex(), getHighlightCellColumnIndex());
+
+        if (!cell.isMerged()) {
+            r.setY(rendererHelper.getRowOffset(visibleRowIndex));
+        } else {
+            int firstRowOfMergedBlock = getFirstRowIndexOfMergedBlock(model, visibleRowIndex);
+            r.setY(rendererHelper.getRowOffset(firstRowOfMergedBlock));
+        }
+    }
+
+    int getFirstRowIndexOfMergedBlock(final GridData model, final int rowIndex) {
+
+        int currentIndex = rowIndex;
+        GridCell<?> cell = model.getCell(rowIndex, getHighlightCellColumnIndex());
+        while (cell.getMergedCellCount() == 0) {
```

Comment:
I think we should have the same condition as we have in `getMergedCellsCount`:
```java
cell.getMergedCellCount() == 0 && currentIndex >= 0
```

Reply:
@danielzhe hello, please let me know what do you think about this comment.

Reply:
Changes applied.